3-async-await/solution.js

```diff
@@ -20,17 +20,63 @@ the format is up to you
 
 8. as an extra challenge add Promise.race() and Promise.any(), and try to get the idea of what happens
 */
+const prices = require('./prices')
+const products = require('./products')
 
-function solution() {
-    // YOUR SOLUTION GOES HERE
 
-    // You generate your id value here
+const idGenerator = () => {
+    const id = Date.now().toString();
+    return parseInt(id.slice(id.length - 2));
+}
+
+const promiseAll = async (id)=> {
+    try{
+    const allPromise = Promise.all([
+        products(id),
+        prices(id),       
+    ])
+    
+        const [product,price] = await allPromise;
+        console.log({id,product,price});
+    }
+    catch(error){
+        console.log(error.message);
+    }
+
+}
+
+const promiseAllSettled = async (id) => {
+
+    const [ product, price ]= await Promise.allSettled([
+
+        products(id),
+        prices(id),
+    ])
+    
+  
+    if (product.reason) {
+        console.log(`PRODUCT: ${product.reason.message}`);
+        
+    }
+    if (price.reason) {
+        console.log(`PRICE: ${price.reason.message}`);
+        
+    }
```

Comment:
Im not sure you really need the if statement here, if you think that the reason is going to be falsy, because of an error, you can just wrap it all in a `try {} catch{}` instead